server/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodeTests.java

```diff
@@ -90,22 +93,20 @@ public void testDiscoveryNodeIsRemoteClusterClientDefault() {
     }
 
     public void testDiscoveryNodeIsRemoteClusterClientSet() {
-        runTestDiscoveryNodeIsRemoteClusterClient(Settings.builder().put(Node.NODE_REMOTE_CLUSTER_CLIENT.getKey(), true).build(), true);
+        runTestDiscoveryNodeIsRemoteClusterClient(remoteClusterClientNode(), true);
     }
 
     public void testDiscoveryNodeIsRemoteClusterClientUnset() {
-        runTestDiscoveryNodeIsRemoteClusterClient(Settings.builder().put(Node.NODE_REMOTE_CLUSTER_CLIENT.getKey(), false).build(), false);
+        runTestDiscoveryNodeIsRemoteClusterClient(nonRemoteClusterClientNode(), false);
     }
 
     private void runTestDiscoveryNodeIsRemoteClusterClient(final Settings settings, final boolean expected) {
         final DiscoveryNode node = DiscoveryNode.createLocal(settings, new TransportAddress(TransportAddress.META_ADDRESS, 9200), "node");
         assertThat(node.isRemoteClusterClient(), equalTo(expected));
-        final Set<DiscoveryNodeRole> expectedRoles = new HashSet<>(DiscoveryNodeRole.BUILT_IN_ROLES);
         if (expected) {
-            assertThat(node.getRoles(), equalTo(expectedRoles));
+            assertThat(node.getRoles(), hasItem(DiscoveryNodeRole.REMOTE_CLUSTER_CLIENT_ROLE));
```

Comment:
I'm not sure about this. The assertion will hold if the roles collection has at least one item equal to the specifed role. That's not the same as it being the only role in the collection.

Reply:
Hmm, actually I may be misreading the before and after. Please ignore if I've misunderstood.

Reply:
The intention is not to test the exact list of roles, rather to test whether or not the node has the remote cluster client role based on whether or not it was in the list of configured roles. Therefore I have modified the test to reflect that.